Automattic/wp-calypso
# Dialog: Use `react-modal` to implement

Rationale: Lets us avoid SSR-incompatible dependencies, mixins, and removes some implementation burden.  `react-modal` seems to be pretty well-maintained and official-ish (lives in the `reactjs` GitHub org).

To test:
* Check that Dialog still works in all (heh) contexts. Some examples:
 * Site Selector Modal: `/design`, clicking any theme action
 * `lib/accept`, e.g. `/design/<jetpacksite>`, delete theme
 * `/media` section/image editor, since it's using one of the more obscure props that are being replaced (see 'Notes' below)
* Specifically, check that
  * Styling is unchanged with respect to `master`/production
  * Dialog buttons work
  * Clicking outside the dialog closes it
  * Hitting Esc closes it

Notes:
* This loses `Dialog`'s `enterTimeout, `transitionLeave`, and `onClickOutside` props.
  * `enterTimeout` wasn't used at all
  * `transitionLeave` defaulted to `true` and was only set (to `false`) by `disconnect-jetpack-dialog.jsx`. I've removed it.
  * `onClickOutside` was used twice
    * one of which was incorrect and without effect AFAICS (`my-sites/media/main.jsx`)
    * I changed the other to use `react-modal`'s `shouldCloseOnOverlayClick` prop, but that still lost a criterion which I hope was only for "outside" detection (`post-editor/media-modal/index.jsx`)
